ftw/simplelayout/browser/anchors.py

```diff
@@ -12,7 +19,36 @@ def listAnchorNames(self, *args, **kwargs):
         query = {'object_provides': ISimplelayoutBlock.__identifier__}
 
         for block in self.context.listFolderContents(contentFilter=query):
-            view = block.restrictedTraverse('content_anchors')
-            anchors.extend(view.listAnchorNames(*args, **kwargs))
+            fields = self.get_html_fields(block)
+
+            for field in fields:
+                anchors.extend(
+                    self.extract_anchors(
+                        field.get(block).output
+                    )
+                )
 
         return anchors
+
+    def get_html_fields(self, obj):
+        fields = []
+        for schema in iterSchemata(obj):
+            for name, field in getFieldsInOrder(schema):
+                # Only consider RichText fields.
+                if IRichText.providedBy(field):
+                    fields.append(field)
+
+        # Filter empty fields (i.e. not having a value).
+        fields = filter(None, fields)
```

Comment:
I think this does not filter "empty fields" as promised, but rather fields that do not exist 🤔 
I would either move the value check to the `listAnchorNames` method or change this method to return values rather than fields.

Is it easily possible to add a test where the field value is actually `None`, so that we could detect that it crashed in this state?